services/slam/builtin/orbslam_int_test.go

```diff
@@ -211,36 +231,55 @@ func integrationTestHelper(t *testing.T, mode slam.Mode) {
 	svc, err = createSLAMService(t, attrCfg, golog.NewTestLogger(t), true, true)
 	test.That(t, err, test.ShouldBeNil)
 
+	// Check if orbslam hangs and needs to be shut down
+	orbslam_hangs = false
+	start_time_sent_image := time.Now()
 	// Wait for orbslam to finish processing images
 	logReader = svc.(internal.Service).GetSLAMProcessBufferedLogReader()
 	for {
 		line, err := logReader.ReadString('\n')
 		test.That(t, err, test.ShouldBeNil)
+		if strings.Contains(line, "Passed image to SLAM") {
+			start_time_sent_image = time.Now()
+		}
 		if strings.Contains(line, "Finished processing offline images") {
 			break
 		}
 		test.That(t, strings.Contains(line, "Fail to track local map!"), test.ShouldBeFalse)
+		if time.Since(start_time_sent_image) > time.Duration(dataInsertionMaxTimeoutMin)*time.Minute {
+			orbslam_hangs = true
+			t.Log("orbslam hangs")
+			break
+		}
 	}
 
 	testOrbslamPositionAndMap(t, svc)
 
-	// Wait for the final map to be saved
-	for {
-		line, err := logReader.ReadString('\n')
-		test.That(t, err, test.ShouldBeNil)
-		if strings.Contains(line, "Finished saving final map") {
-			break
+	if !orbslam_hangs {
+		// Wait for the final map to be saved
+		for {
+			line, err := logReader.ReadString('\n')
+			test.That(t, err, test.ShouldBeNil)
+			if strings.Contains(line, "Finished saving final map") {
+				break
+			}
 		}
 	}
 
 	// Close out slam service
-	test.That(t, utils.TryClose(context.Background(), svc), test.ShouldBeNil)
+	err = utils.TryClose(context.Background(), svc)
```

Comment:
[q] why isnt this combined with the !orbslam_hang check above?

Reply:
Because we'd be duplicating the `err = utils.TryClose(context.Background(), svc)` line if I'd combine it, and because the code is written in sequential order and I think this choice fits the existing style best.

Reply:
I see, you want the closeout to happen no matter the value of orbslam_hangs